lib/mox.ex

```diff
@@ -170,6 +170,48 @@ defmodule Mox do
 
       setup :set_mox_from_context
 
+  ### Blocking on expectations
+
+  If your mock is called in a different process than the test process,
+  in some cases there is a chance that the test will finish executing
+  before it has a chance to call the mock and meet the expectations.
+  Imagine this:
+
+      test "calling a mock from a different process" do
+        expect(MyApp.CalcMock, :add, fn x, y -> x + y end)
+
+        spawn(fn -> MyApp.CalcMock.add(4, 2) end)
+
+        verify!()
+      end
+
+  The test above has a race condition because there is a chance that the
+  `verify!/0` call will happen before the spawned process calls the mock.
+  In most cases, you don't control the spawning of the process so you can't
+  simply monitor the process to know when it dies in order to avoid this
+  race condition. In those cases, the way to go is to "sync up" with the
+  process that calls the mock by sending a message to the test process
+  from the expectation and using that to know when the expectation has been
+  called.
+
+      test "calling a mock from a different process" do
+        parent = self()
+        ref = make_ref()
+
+        expect(MyApp.CalcMock, :add, fn x, y ->
+          send(parent, {ref, :add})
```

Comment:
Sending just the ref would be enough, no?

Reply:
Well yeah but then I would have to name it `add_ref` or something cause I'm afraid people would then use the same ref for all their blocking expectations, defeating the purpose... 😄